src/bin/main.rs

```diff
@@ -1,22 +1,41 @@
-use std::thread;
+use altctrl::{gui, i2c, AltctrlInterface, Event};
+use clap::{App, Arg};
 use std::sync::mpsc;
-use altctrl::{i2c, gui, Event, AltctrlInterface};
+use std::thread;
 
 // Main thread of altctrl
 // Handles the setup of all the modules in the project
 // Handles message schedule
 fn main() {
-    let interface = std::env::args().nth(0).expect("should get first argument");
-
-    let interface: Box<dyn AltctrlInterface + Send> = match &*interface {
-        "chungo" => Box::new(altctrl::Chungo),
-        "garfanzo" => Box::new(altctrl::Garfanzo),
-        "fatkhiyev" => Box::new(altctrl::Fatkhiyev),
-        _ => {
-            println!("Invalid interface!");
-            return;
-        }
-    };
+    let matches = App::new("altctrl")
+        .arg(
+            Arg::with_name("interface")
+                .short("i")
+                .long("interface")
+                .help("Set the interface for altctl.")
+                .takes_value(true)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("disable_i2c")
+                .short("d")
+                .long("disable_i2c")
+                .help("Disable i2c module."),
+        )
+        .get_matches();
+
+    let interface: Box<dyn AltctrlInterface + Send> =
+        match matches.value_of("interface").unwrap().as_ref() {
+            "chungo" => Box::new(altctrl::Chungo),
+            "garfanzo" => Box::new(altctrl::Garfanzo),
+            "fatkhiyev" => Box::new(altctrl::Fatkhiyev),
+            _ => {
+                println!("Invalid interface!");
+                return;
+            }
+        };
```

Comment:
Here, if the user fails to provide an `interface` flag, the program will crash rather than exit gracefully. You should remove the unwrap call here and instead build the Option variants into your match, something like this:

```rust
match matches.value_of("interface").as_ref() {
    Some("chungo") => ...
    Some("garfanzo") => ...
    Some("fatkhiyev") => ...
    Some(_) | None => {
        println!("Invalid interface (or no interface provided)");
        return;
    }
}
```

Another thing you could do is decide to provide a "default" interface by declaring a default value as part of the Clap definition, in which case the unwrap would be safe.

---

Looking at this again, you marked `interface` as required, so the unwrap might not be a problem to begin with. But I'm going to leave this comment here to let you know that default values are possible.